lib/std/fs.zig

```diff
@@ -938,10 +938,10 @@ pub const Dir = struct {
         const w = os.wasi;
         var fdflags: w.fdflags_t = 0x0;
         var base: w.rights_t = 0x0;
-        if (flags.read) {
+        if (flags.isRead()) {
```

Comment:
In hindsight, I could just have done `flags.mode != .write_only`. Would that be better? Or less legible?